### PR TITLE
Added Exportable bits to T5 + unit test

### PR DESCRIPTION
**Pull request by borisfom**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by borisfom** [comment]
Here's the patch that needs to be applied to transformers for ONNX export of T5 to work : 
--- /usr/local/lib/python3.10/dist-packages/transformers/models/t5/modeling_t5.orig.py  2023-08-30 19:14:16.764941813 +0000
+++ /usr/local/lib/python3.10/dist-packages/transformers/models/t5/modeling_t5.py       2023-08-30 20:05:44.435873449 +0000
@@ -425,12 +425,12 @@
             torch.log(relative_position.float() / max_exact)
             / math.log(max_distance / max_exact)
             * (num_buckets - max_exact)
-        ).to(torch.long)
+        )
         relative_position_if_large = torch.min(
             relative_position_if_large, torch.full_like(relative_position_if_large, num_buckets - 1)
         )

-        relative_buckets += torch.where(is_small, relative_position, relative_position_if_large)
+        relative_buckets += torch.where(is_small, relative_position.to(torch.long), relative_position_if_large.to(torch.long))
         return relative_buckets

     def compute_bias(self, query_length, key_length, device=None):
@@ -851,7 +851,7 @@
     def _shift_right(self, input_ids):
         decoder_start_token_id = self.config.decoder_start_token_id
         pad_token_id = self.config.pad_token_id
-
+        input_ids=input_ids.to(dtype=torch.long)
         if decoder_start_token_id is None:
             raise ValueError(
                 "self.model.config.decoder_start_token_id has to be defined. In T5 it is usually set to the pad_token_id."
@@ -861,7 +861,7 @@
         # shift inputs to the right
         if is_torch_fx_proxy(input_ids):
             # Item assignment is not supported natively for proxies.
-            shifted_input_ids = torch.full(input_ids.shape[:-1] + (1,), decoder_start_token_id)
+            shifted_input_ids = input_ids.new_full(input_ids.shape[:-1] + (1,), decoder_start_token_id)
             shifted_input_ids = torch.cat([shifted_input_ids, input_ids[..., :-1]], dim=-1)
         else:
             shifted_input_ids = input_ids.new_zeros(input_ids.shape)
@@ -979,7 +979,7 @@
             )
         elif input_ids is not None:
             input_shape = input_ids.size()
-            input_ids = input_ids.view(-1, input_shape[-1])
+            input_ids = input_ids.to(torch.long).view(-1, input_shape[-1])
         elif inputs_embeds is not None:
             input_shape = inputs_embeds.size()[:-1]
         else:
